### PR TITLE
Implement higher order reducers

### LINKED ISSUE #503: Missing Arithmetic reductions
- [x] sem
- [x] product (only DataFrame)
- [x] quantile
- [x] skew
- [x] kurtosis
- [x] describe

### PR DESCRIPTION
**Pull request by phofl**
ref #525

this sits on the other 2 prs

closes #503

### CODE REVIEW

**Comment by milesgranger** [comment]
> this sits on the other 2 prs

I'm assuming #708 and #709? Will looks at again when updated after both are in. :+1:

**Comment by phofl** [comment]
rebased

**Review by milesgranger — APPROVED** [review]
Super, another nit, but looks good for me!

File: dask_expr/_core.py [hunk 1]
```diff
@@ -438,7 +438,10 @@ def __dask_graph__(self):
                 continue
             seen.add(expr._name)
 
-            layers.append(expr._layer())
+            try:
+                layers.append(expr._layer())
+            except Exception:
+                raise
```
[the remark this answers is not in the document]
Comment (phofl) [reply dask_expr/_core.py]: bad rebase, sorry

**Pull request merged**